### ISSUE TITLE
[Accessibility issue] Some links and text buttons cannot be easily identified for users with color deficiencies. 

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#73396]
**Steps to reproduce**:
Go to [play.grafana.org/alerting](https://play.grafana.org/alerting). Simulate a color deficiency using your browser tools (e.g. Firefox accessibility tab > Simulate; Chrome devtools > More tools > Rendering > Emulate vision deficiencies). Notice how difficult it is to discern what is a link.

**Actual Result**:
Links, text buttons and other interactive controls are often difficult to identify for users with vision deficiencies.

**Expected Result**
Interactive controls should be easily recognisable without color and have some sort of designator they are interactive. e.g. links have underlines, a chevron or another common design pattern to discern them.

![image](https://github.com/grafana/grafana/assets/6906380/9046b1c6-1bfd-438e-8acb-770bcbdd8b62)

UHG Audit ID: 1.401

**Comment by ashharrison90** [comment]
our `TextLink` component provides an `inline` prop which shows an underline. either we can:
- ensure this use case is using `inline`
- think about a different way to show non-inline, non-external links (external links show an icon which makes it clear as well)
